services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationService.kt

```diff
@@ -28,19 +32,25 @@ private val logger = KotlinLogging.logger {}
 const val ALLU_APPLICATION_ERROR_MSG = "Error sending application to Allu"
 
 open class ApplicationService(
-    private val repo: ApplicationRepository,
+    private val applicationRepository: ApplicationRepository,
     private val alluStatusRepository: AlluStatusRepository,
     private val cableReportService: CableReportService,
     private val disclosureLogService: DisclosureLogService,
     private val applicationLoggingService: ApplicationLoggingService,
     private val geometriatDao: GeometriatDao,
+    private val permissionService: PermissionService,
+    private val hankeRepository: HankeRepository,
 ) {
+    @Transactional
     open fun getAllApplicationsForUser(userId: String): List<Application> {
-        return repo.getAllByUserId(userId).map { it.toApplication() }
+        val result = applicationRepository.getAllByUserId(userId).toMutableSet()
+        val hankkeet =
+            permissionService.getAllowedHankeIds(userId = userId, permission = PermissionCode.VIEW)
+        hankkeet.forEach { result.addAll(hankeRepository.getOne(it).hakemukset) }
+        return result.map { it.toApplication() }
```

Comment:
I think we can only return the applications with matching permissions. They should include the ones they have created, unless they are explicitly removed from the hanke.

Also, it's more efficient to get the hankkeet in a single query.

```suggestion
        val hankeIds =
            permissionService.getAllowedHankeIds(userId = userId, permission = PermissionCode.VIEW)
        return hankeRepository
            .findAllById(hankeIds)
            .flatMap { it.hakemukset }
            .map { it.toApplication() }
```

Reply:
Done.